draft-ietf-add-split-horizon-authority.xml

```diff
@@ -179,158 +185,163 @@
+            the named local resolver (potentially including the entire parent
+            zone).</li>
+        <li>An NSEC3 Hash Algorithm.</li>
+        <li>A high-entropy salt.</li>
+      </ul>
+      <t>If the local encrypted resolver is identified by name (e.g., DNR), that
+      identifying name MUST be the one used in any corresponding Authorization
+      Claim.  Otherwise (e.g., DDR using IP addresses), the resolver MUST
+      present a validatable certificate containing a subjectAltName that
+      matches the Authorization Claim.</t>
+      <t>To establish its authority, the network MUST provide each Authorization
+      Claim to the parent zone operator.  If the contents are approved, the
+      parent zone operator computes a "Verification Token" according to the
+      following procedure based on <relref section="5" target="RFC5155"/>:</t>
+      <ol>
+        <li>Convert all subdomains into canonical form.</li>
+        <li>Replace the suffix corresponding to the parent zone with a zero
+            byte.</li>
+        <li>Let $X be the concatenation of the resulting pseudo-FQDNs.</li>
+        <li>Let $TOKEN = IH($SALT, $X, 0)</li>
+      </ol>
+      <t>The zone operator then publishes a "Verification Record" with the
+      following structure, following the advice of
+      <xref target="I-D.ietf-dnsop-domain-verification-techniques"/>:</t>
+      <ul>
+        <li>Type = TXT.</li>
+        <li>Owner Name = Concatenation of the ADN, "_splitdns-challenge", and
+            the parent zone name.</li>
+        <li>Contents = "token=base64url($TOKEN)".</li>
+      </ul>
+      <t>By publishing this record, the parent zone authorizes the local
+      encrypted resolver to serve these subdomains authoritatively.</t>
+      <section>
+        <name>Example</name>
+        <t>Consider the following authorization claim:</t>
+        <ul>
+          <li>ADN = "resolver17.parent.example"</li>
+          <li>Parent = "parent.example"</li>
+          <li>Subdomains = "payroll.parent.example",
+              "secret.project.parent.example"</li>
+          <li>Hash Algorithm = SHA-1</li>
+          <li>Salt = "example salt bytes (should be random)"</li>
+        </ul>
+        <t>To approve this claim, the zone operator would publish the following record:</t>
+        <figure><artwork><![CDATA[
+resolver17.parent.example._splitdns-challenge.parent.example. IN TXT \
+"token=kfHudyzZkdQccYVLV0zWsY4ETZUK"
+]]></artwork></figure>
+      </section>
+      <section>
+        <name>Conveying Authorization Claims</name>
+        <t>
+          The Authorization Claim is an abstract structure that must be encoded in
+          some concrete syntax in order to convey it from the network to the client.
+          This section defines some encodings of the Authorization Claims.
+        </t>
+        <section>
+          <name>Using DHCP</name>
+          <t>
+            In DHCP, each Authorization Claim is encoded as a DHCP Authentication
```

Comment:
Why not define a new DHCP option ? 
What is the advantage of extending DHCP Authentication option ?

Reply:
It requires a provisioning step, a secret must be configured on both the DHCP server and the client endpoints. If provisioning involved, the endpoint can also be configured with split-dns configuration. 

I see several unassigned values https://www.iana.org/assignments/bootp-dhcp-parameters/bootp-dhcp-parameters.xhtml#options and we recently got 164 for OPTION_V4_DNR.

Reply:
> > It requires a provisioning step, a secret must be configured on both the DHCP server and the client endpoints.
> 
> No, provisioning is not required. The DHCP server provides this authentication information, and the DHCP client validates it using the mechanism in this draft.

Got it, Thanks. We should say RDM will be set to a zero value (see https://www.rfc-editor.org/rfc/rfc8415.html#section-20.4.1).

Reply:
OK, done.